spec.emu

```diff
@@ -132,6 +920,9 @@ contributors: Luca Casonato, Guy Bedford, Nicolò Ribaudo
             1. <ins>If _moduleSourceObject_ is *null*, then</ins>
               1. <ins>Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; a newly created *TypeError* object &raquo;).</ins>
               1. <ins>Return ~unused~.</ins>
+            1. <ins>Assert: _moduleSourceObject_.[[Prototype]] is %AbstractModuleSourcePrototype%.</ins>
```

Comment:
I think the hierarchy here is wrong. It's asserting that
```js
import source s from "./mod.wasm";
Object.getPrototypeOf(s) === %AbstractModuleSourcePrototype%;
```

what I would expect is that every module source type has its own prototype class, and they all inherit from `%AbstractModuleSourcePrototype%`:

```js
import source s from "./mod.wasm";

// These equalities are already true with the current WebAssembly.Module class
s instanceof WebAssembly.Module;
Object.getPrototypeOf(s) === WebAssembly.Module.prototype;
s.constructor === WebAssembly.Module
Object.hasOwn(s, "constructor") === false;
Object.hasOwn(Object.getPrototypeOf(s), "constructor") === true;

// These equalities are new
Object.getPrototypeOf(Object.getPrototypeOf(s) === %AbstractModuleSourcePrototype%;
s instanceof Object.assign(function(){}, { prototype: %AbstractModuleSourcePrototype% })
```

i.e. it should be like this:
```js
WebAssembly.Module = function () { ... }
Object.setPrototypeOf(WebAssembly.Module, %AbstractModuleSourcePrototype%);
```
and not like this:
```js
WebAssembly.Module = function () { ... }
WebAssembly.Module.prototype = %AbstractModuleSourcePrototype%;
```

---

EDIT: What I'm suggesting is already what you wrote in the "The ModuleSource Constructor" section. It's just wrong in this assertion, where it should probably be

```suggestion
            1. <ins>Assert: _moduleSourceObject_.[[Prototype]].[[Prototype]] is %AbstractModuleSourcePrototype%.</ins>
```

Reply:
Fixed, thanks.